### ISSUE TITLE
ColumnPointTable Destructor Slow

### ISSUE THREAD

**Issue by b-gutzmann** [PDAL/PDAL#3990]
https://github.com/PDAL/PDAL/blob/366475794646082b973fc5ce3e142aa3aaec8a23/pdal/PointTable.hpp#L186

The ColumnPointTable destructor is slow with large point clouds because of the amount of buffers it allocates in addPoint(). Changing m_blockPtCnt from 16384 to 65536 (what RowPointTable uses) helps. Maybe a more complex approach where the buffer size grows over time is warranted.

I'm testing with tiffs found [here](https://visibleearth.nasa.gov/images/57752/blue-marble-land-surface-shallow-water-and-shaded-topography) and a simple pipeline that just reads the data in standard mode. With the largest tiff (21600 x 21600), the destructor took 78.713 seconds. After changing m_blockPtCnt to 65536, the destructor took 18.106 seconds. With (8192 x 4096), 146.3 versus 69.4 ms.
Timings were taken on windows in release mode.

`[
    "C:/test_data/land_shallow_topo_east.tif"
]`
`pdal.exe pipeline --nostream "pipeline_basic.json"`

**Comment by hobu** [comment]
What was the pipeline you were using?

**Comment by b-gutzmann** [comment]
Just 
```
[ 
    "C:/test_data/land_shallow_topo_8192.tif" 
]
```
Same results with 
```
[
    "C:/test_data/land_shallow_topo_8192.tif",
    "C:/test_data/temp.laz"
]
````

**Comment by abellgithub** [comment]
This seems reasonable. Since there's are blocks for each dimension, they should probably be pretty large.

**Comment by abellgithub** [comment]
These timings seem really high (slow). I just haven't seen timings like that. Are things built in release mode? Something else going on?

**Comment by b-gutzmann** [comment]
My memory usage with the large dataset (466 560 000 points) was getting quite high. However I had no such issue with the smaller dataset (33 554 432 points). Using Measure-Command against my build produces the same timings as the conda release of pdal, so I don't think the issue is my compiler/configuration.

In general the GDALReader is slow which is what I was originally looking into. This is a small piece of that. But that's a separate issue as this issue isn't GDALReader specific.

**Comment by hobu** [comment]
> GDALReader is slow

`readers.gdal` is mostly just a toy. What were you using it for? It is going to be quite sensitive to `GDAL_CACHE_MAX` too, so you might look at bumping that for better performance.

**Comment by b-gutzmann** [comment]
Reading GeoTIFF elevation data, cropping it, and making a mesh

**Comment by abellgithub** [comment]
Can you stick the file someplace it can be downloaded?

**Comment by b-gutzmann** [comment]
The GeoTIFFs? I included a link in the original comment. [https://visibleearth.nasa.gov/images/57752/blue-marble-land-surface-shallow-water-and-shaded-topography](https://visibleearth.nasa.gov/images/57752/blue-marble-land-surface-shallow-water-and-shaded-topography) [https://eoimages.gsfc.nasa.gov/images/imagerecords/57000/57752/land_shallow_topo_8192.tif](https://eoimages.gsfc.nasa.gov/images/imagerecords/57000/57752/land_shallow_topo_8192.tif)

**Comment by hobu** [comment]
Another thing to try is `readers.gdal.memorycopy=true`, which will load the entire raster up in to memory first. That won't fix the ColumnPointTable dtor speed issue, but it might make the read a lot faster if you are reading the entire raster anyway.

**Comment by hobu** [comment]
Clsoing this as I think most of the issues were in reading raster data which were addressed in #3999 and #4001. Please reopen if there is still an issue.
